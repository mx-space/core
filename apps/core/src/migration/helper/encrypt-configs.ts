import { plainToInstance } from 'class-transformer'

import 'reflect-metadata'

import { ENCRYPT } from '~/app.config'
import { register } from '~/global/index.global'
import { generateDefaultConfig } from '~/modules/configs/configs.default'
import * as optionDtos from '~/modules/configs/configs.dto'
import { encryptObject } from '~/modules/configs/configs.encrypt.util'
import { IConfig, IConfigKeys } from '~/modules/configs/configs.interface'
import { getDatabaseConnection } from '~/utils/database.util'

console.log(ENCRYPT)

const allOptionKeys: Set<IConfigKeys> = new Set()
Object.entries(optionDtos).reduce((obj, [key, value]) => {
  const optionKey = (key.charAt(0).toLowerCase() +
    key.slice(1).replace(/Dto$/, '')) as IConfigKeys
  allOptionKeys.add(optionKey)
  return {
    ...obj,
    [`${optionKey}`]: value,
  }
}, {})

async function main() {
  await register()
  const connection = await getDatabaseConnection()
  const db = connection.db
  const configs: any[] = []
  await db
    .collection('options')
    .find()
    .forEach((current) => {
      configs.push(current)
    })

  const mergedConfig = generateDefaultConfig()
  configs.forEach((field) => {
    const name = field.name as keyof IConfig

    if (!allOptionKeys.has(name)) {
      return
    }

    const value = field.value
    mergedConfig[name] = { ...mergedConfig[name], ...value }
  })

  const encrypted = encryptObject(plainToInstance(IConfig as any, mergedConfig))

  for await (const [key, value] of Object.entries(encrypted)) {
    configs[key] = value
    await db.collection('options').updateOne(
      {
        name: key,
      },
      {
        $set: {
          value,
        },
      },
    )
  }

  await connection.close()
  process.exit(0)
}

main()
