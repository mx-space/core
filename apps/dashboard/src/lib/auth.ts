import { getStorageNS } from './ns'

const CANDIDATE_TOKEN_KEYS = [
  'token',
  getStorageNS('token'),
  'auth:token',
  getStorageNS('auth:token'),
]

export function getAuthToken(): string | null {
  try {
    for (const key of CANDIDATE_TOKEN_KEYS) {
      const value = localStorage.getItem(key)
      if (value) return value
    }
  } catch {
    // ignore storage access errors (e.g., SSR or privacy mode)
  }
  return null
}

export function setAuthToken(token: string) {
  try {
    localStorage.setItem(getStorageNS('token'), token)
  } catch {
    // ignore storage write errors
  }
}

export function getSocketAuth(): { token?: string } {
  const token = getAuthToken()
  if (!token) return {}
  const normalized =
    token.startsWith('bearer ') || token.startsWith('Bearer ')
      ? token
      : `bearer ${token}`
  return { token: normalized }
}
