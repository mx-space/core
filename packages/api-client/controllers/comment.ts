import type { IRequestAdapter } from '~/interfaces/adapter'
import type { IController } from '~/interfaces/controller'
import type { PaginationParams } from '~/interfaces/params'
import type { IRequestHandler } from '~/interfaces/request'
import type { ReaderModel } from '~/models'
import type { PaginateResult } from '~/models/base'
import type { CommentModel } from '~/models/comment'
import type { HTTPClient } from '../core'
import type { CommentDto } from '../dtos/comment'

import { autoBind } from '~/utils/auto-bind'

declare module '../core/client' {
  interface HTTPClient<
    T extends IRequestAdapter = IRequestAdapter,
    ResponseWrapper = unknown,
  > {
    comment: CommentController<ResponseWrapper>
  }
}

export class CommentController<ResponseWrapper> implements IController {
  base = 'comments'
  name = 'comment'

  constructor(private readonly client: HTTPClient) {
    autoBind(this)
  }

  get proxy(): IRequestHandler<ResponseWrapper> {
    return this.client.proxy(this.base)
  }

  /**
   * 根据 comment id 获取评论，包括子评论
   */
  getById(id: string) {
    return this.proxy(id).get<CommentModel & { ref: string }>()
  }

  /**
   * 获取文章的评论列表
   * @param refId 文章 Id
   */
  getByRefId(refId: string, pagination: PaginationParams = {}) {
    const { page, size } = pagination
    return this.proxy.ref(refId).get<
      PaginateResult<CommentModel & { ref: string }> & {
        readers: Record<string, ReaderModel>
      }
    >({
      params: { page: page || 1, size: size || 10 },
    })
  }
  /**
   * 评论
   */
  comment(refId: string, data: CommentDto) {
    return this.proxy(refId).post<CommentModel>({
      data,
    })
  }

  /**
   * 回复评论
   */
  reply(commentId: string, data: CommentDto) {
    return this.proxy.reply(commentId).post<CommentModel>({
      data,
    })
  }
}
