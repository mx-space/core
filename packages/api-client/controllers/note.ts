import type { IRequestAdapter } from '~/interfaces/adapter'
import type { IController } from '~/interfaces/controller'
import type { IRequestHandler, RequestProxyResult } from '~/interfaces/request'
import type { SelectFields } from '~/interfaces/types'
import type { PaginateResult } from '~/models/base'
import type {
  NoteModel,
  NoteWrappedPayload,
  NoteWrappedWithLikedPayload,
} from '~/models/note'
import type { HTTPClient } from '../core/client'
import type { SortOptions } from './base'

import { autoBind } from '~/utils/auto-bind'

declare module '../core/client' {
  interface HTTPClient<
    T extends IRequestAdapter = IRequestAdapter,
    ResponseWrapper = unknown,
  > {
    note: NoteController<ResponseWrapper>
  }
}

export type NoteListOptions = {
  select?: SelectFields<keyof NoteModel>
  year?: number
  sortBy?: 'weather' | 'mood' | 'title' | 'created' | 'modified'
  sortOrder?: 1 | -1
}

export class NoteController<ResponseWrapper> implements IController {
  base = 'notes'
  name = 'note'

  constructor(private client: HTTPClient) {
    autoBind(this)
  }
  get proxy(): IRequestHandler<ResponseWrapper> {
    return this.client.proxy(this.base)
  }

  /**
   * 最新日记
   */
  getLatest() {
    return this.proxy.latest.get<NoteWrappedWithLikedPayload>()
  }

  /**
   * 获取一篇日记，根据 Id 查询需要鉴权
   * @param id id | nid
   * @param password 访问密码
   */

  getNoteById(
    id: string,
  ): Promise<RequestProxyResult<NoteModel, ResponseWrapper>>
  getNoteById(id: number): Promise<NoteWrappedPayload>
  getNoteById(id: number, password: string): Promise<NoteWrappedPayload>
  getNoteById(
    id: number,
    password: undefined,
    singleResult: true,
  ): Promise<RequestProxyResult<NoteModel, ResponseWrapper>>
  getNoteById(
    id: number,
    password: string,
    singleResult: true,
  ): Promise<RequestProxyResult<NoteModel, ResponseWrapper>>
  getNoteById(...rest: any[]): any {
    const [id, password = undefined, singleResult = false] = rest

    if (typeof id === 'number') {
      return this.proxy.nid(id.toString()).get<NoteWrappedWithLikedPayload>({
        params: { password, single: singleResult ? '1' : undefined },
      })
    } else {
      return this.proxy(id).get<NoteModel>()
    }
  }

  /**
   * 日记列表分页
   */

  getList(page = 1, perPage = 10, options: NoteListOptions = {}) {
    const { select, sortBy, sortOrder, year } = options
    return this.proxy.get<PaginateResult<NoteModel>>({
      params: {
        page,
        size: perPage,
        select: select?.join(' '),
        sortBy,
        sortOrder,
        year,
      },
    })
  }

  /**
   * 获取当前日记的上下各 n / 2 篇日记
   */
  getMiddleList(id: string, size = 5) {
    return this.proxy.list(id).get<{
      data: Pick<
        NoteModel,
        'id' | 'title' | 'nid' | 'created' | 'isPublished'
      >[]
      size: number
    }>({
      params: { size },
    })
  }

  /**
   * 获取专栏内的所有日记
   */
  getNoteByTopicId(
    topicId: string,
    page = 1,
    size = 10,
    sortOptions: SortOptions = {},
  ) {
    return this.proxy.topics(topicId).get<PaginateResult<NoteModel>>({
      params: { page, size, ...sortOptions },
    })
  }
}
