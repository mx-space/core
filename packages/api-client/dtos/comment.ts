export interface CommentDto {
  author: string

  text: string

  mail: string

  url?: string

  source?: 'github' | 'google'
  avatar?: string
}
