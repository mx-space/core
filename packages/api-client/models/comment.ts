import type { BaseModel } from './base'
import type { CategoryModel } from './category'

export enum RefType {
  Page = 'Page',
  Post = 'Post',
  Note = 'Note',
}
export interface CommentModel extends BaseModel {
  refType: RefType
  ref: string
  state: number
  commentsIndex: number
  author: string
  text: string
  mail?: string
  url?: string
  ip?: string
  agent?: string
  key: string
  pin?: boolean

  avatar: string

  parent?: CommentModel | string
  children: CommentModel[]

  isWhispers?: boolean
  location?: string

  source?: string
}
export interface CommentRef {
  id: string
  categoryId?: string
  slug: string
  title: string
  category?: CategoryModel
}

export enum CommentState {
  Unread,
  Read,
  Junk,
}
