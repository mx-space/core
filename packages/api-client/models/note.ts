import type { ModelWithLiked, TextBaseModel } from './base'
import type { TopicModel } from './topic'

export interface NoteModel extends TextBaseModel {
  isPublished: boolean
  count: {
    read: number
    like: number
  }

  mood?: string
  weather?: string
  bookmark?: boolean

  publicAt?: Date
  password?: string | null
  nid: number

  location?: string

  coordinates?: Coordinate
  topic?: TopicModel
  topicId?: string
}

export interface Coordinate {
  latitude: number
  longitude: number
}

export interface NoteWrappedPayload {
  data: NoteModel
  next?: Partial<NoteModel>
  prev?: Partial<NoteModel>
}

export interface NoteWrappedWithLikedPayload {
  data: ModelWithLiked<NoteModel>
  next?: Partial<NoteModel>
  prev?: Partial<NoteModel>
}
