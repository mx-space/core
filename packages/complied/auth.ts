import * as AuthCore from '@auth/core'
import * as AuthCoreAdapters from '@auth/core/adapters'
import * as AuthCoreErrors from '@auth/core/errors'
import * as AuthCoreGithub from '@auth/core/providers/github'
import * as AuthCoreGoogle from '@auth/core/providers/google'

export const authjs = {
  ...AuthCore,
  ...AuthCoreAdapters,

  ...AuthCoreErrors,
  providers: {
    google: AuthCoreGoogle.default,
    github: AuthCoreGithub.default,
  },
}

export type * from '@auth/core/errors'
export type * from '@auth/core/types'
export type * from '@auth/core/providers/google'
export type * from '@auth/core/providers/github'
export * from '@auth/core'

export type * from '@auth/core/adapters'

export { MongoDBAdapter } from '@auth/mongodb-adapter'
