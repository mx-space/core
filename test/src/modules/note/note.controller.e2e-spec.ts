import { createE2EApp } from 'test/helper/create-e2e-app'
import { authProvider } from 'test/mock/modules/auth.mock'
import { configProvider } from 'test/mock/modules/config.mock'
import { gatewayProviders } from 'test/mock/modules/gateway.mock'
import { countingServiceProvider } from 'test/mock/processors/counting.mock'

import { EventEmitter2 } from '@nestjs/event-emitter'
import { ReturnModelType } from '@typegoose/typegoose'

import { CommentService } from '~/modules/comment/comment.service'
import { OptionModel } from '~/modules/configs/configs.model'
import { NoteController } from '~/modules/note/note.controller'
import { NoteModel } from '~/modules/note/note.model'
import { NoteService } from '~/modules/note/note.service'
import { UserModel } from '~/modules/user/user.model'
import { UserService } from '~/modules/user/user.service'
import { CountingService } from '~/processors/helper/helper.counting.service'
import { EventManagerService } from '~/processors/helper/helper.event.service'
import { HttpService } from '~/processors/helper/helper.http.service'
import { ImageService } from '~/processors/helper/helper.image.service'
import { TextMacroService } from '~/processors/helper/helper.macro.service'
import { SubPubBridgeService } from '~/processors/redis/subpub.service'

import MockDbData from './note.e2e-mock.db'

describe('NoteController (e2e)', () => {
  const proxy = createE2EApp({
    controllers: [NoteController],
    providers: [
      NoteService,
      ImageService,
      EventManagerService,
      {
        provide: CommentService,
        useValue: {},
      },

      {
        provide: TextMacroService,
        useValue: {
          async replaceTextMacro(text) {
            return text
          },
        },
      },
      HttpService,
      configProvider,
      EventEmitter2,
      UserService,
      SubPubBridgeService,
      ...gatewayProviders,
      authProvider,
      CountingService,
      countingServiceProvider,
    ],
    imports: [],
    models: [NoteModel, OptionModel, UserModel],
    async pourData(modelMap) {
      // @ts-ignore
      const { model } = modelMap.get(NoteModel) as {
        model: ReturnModelType<typeof NoteModel>
      }
      const documents = await model.insertMany(MockDbData)

      return async () => {
        return documents.map((doc) => doc.remove())
      }
    },
  })

  test('GET /notes', async () => {
    const { app } = proxy
    const res = await app.inject({
      method: 'GET',
      url: '/notes',
    })
    const data = res.json()
    expect(res.statusCode).toBe(200)

    data.data.forEach((d) => {
      delete d.id
      delete d._id
    })
    expect(data).toMatchSnapshot()
  })
})
