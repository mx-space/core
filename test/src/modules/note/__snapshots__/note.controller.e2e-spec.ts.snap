// Vitest Snapshot v1

exports[`NoteController (e2e) > GET /notes 1`] = `
{
  "data": [
    {
      "allow_comment": true,
      "comments_index": 0,
      "count": {
        "like": 0,
        "read": 0,
      },
      "created": "2021-03-01T00:00:00.000Z",
      "hide": false,
      "images": [],
      "meta": null,
      "modified": "2021-03-01T00:00:00.000Z",
      "music": [],
      "nid": 1,
      "text": "Content 1",
      "title": "Note 1",
      "topic": null,
    },
  ],
  "pagination": {
    "current_page": 1,
    "has_next_page": false,
    "has_prev_page": false,
    "size": 10,
    "total": 1,
    "total_page": 1,
  },
}
`;
